### PR TITLE
Add Analytics Zoo cluster serving demo

### PR DESCRIPTION
**Pull request by CharleneHu94**

### CODE REVIEW

**Comment by guzongmin** [comment]
@CharleneHu94 very appreciate your contribution. It is really a well documented demo. Can you please add it into the CI list too?
You could add it into [.github/workflows/demo_test.yml](https://github.com/occlum/occlum/blob/master/.github/workflows/demo_test.yml), which would keep the demo health for each new commit.
When you add the demo into the demo list [demos/README.md](https://github.com/occlum/occlum/blob/master/demos/README.md), users could easily find this demo.

This is the pytorch demo PR for your reference https://github.com/occlum/occlum/commit/f534017d79cd48583d428fb5e740794dbde455bb

File: .github/workflows/demo_test.yml [hunk 1]
```diff
@@ -503,6 +503,31 @@ jobs:
...
+
+    - name: Build dependencies
+      run: docker exec cluster_serving_test bash -c "cd /root/occlum; make submodule"
+
+    - name: Make install
+      run: docker exec cluster_serving_test bash -c "source /opt/intel/sgxsdk/environment; cd /root/occlum; OCCLUM_RELEASE_BUILD=1 make install"
+
+    - name: Set up environment
+      run: docker exec cluster_serving_test bash -c "cd /root/occlum/demos/cluster_serving; source ./environment.sh; ./install-dependencies.sh"
+
```
Comment (guzongmin) [inline .github/workflows/demo_test.yml:527]: In the README file, "Start Flink" is the second step.

Comment (CharleneHu94) [reply .github/workflows/demo_test.yml:527]: In the `README`, there are four major steps: 1) Set up environment; 2) Start Cluster Serving; 3) Push inference image; 4) Stop Cluster Serving. In the `demo_test.yml`, we run step 1) first at `line 525`, then run step 2) and 3) at `line 528`. 

"Start Flink" is the second substep in "2) Start Cluster Serving" if users choose to start components separately, while we do not use this approach in CI test.

**Comment by guzongmin** [comment]
@CharleneHu94 It is an excellent demo, every is perfect.  Can you please merge the 5 commits into a meaningful one? After that, I will merge this PR.

Thanks for your contribution.

**Comment by CharleneHu94** [comment]
> 
> 
> @CharleneHu94 It is an excellent demo, everything is perfect. Can you please merge the 5 commits into a meaningful one? After that, I will merge this PR.
> 
> Thanks for your contribution.

Done :)

**Review by guzongmin — APPROVED** [review]
LGTM

**Pull request merged**